File: cpp/twoSum.cpp
// Two Sum

#include <vector>
#include <unordered_map>
using namespace std;
// Remove these 2 lines when submitting onto leetcode

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numMap;
        // creates a hash map
        int n = nums.size();


        for (int i = 0; i < n; i ++){
            int complement = target - nums[i];
            if (numMap.count(complement)){
                return {numMap[complement], i};
            }
            numMap[nums[i]] = i;
        }
        return {};
    }
};

// Working solution

File: python/twoSum.py
# Two Sum

class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        for i in range(len(nums)):
            if (target - nums[i]) in nums:
                j = nums.index(target - nums[i])
                if j == i:
                    continue
                else:
                    return [i,j]

# Working solution